packages/nimble-components/src/wafer-map/specs/README.md

```diff
@@ -0,0 +1,250 @@
+    -   `orientation` - represent the orientation of the notch on the wafer map outline (only visual). As only four static orientations are possible, it can be represented by an Enum with the following values: top, bottom, left, right.
+    -   `colorScale` - represents the color spectrum which shows the status of the dies on the wafer.\
+        The object we use for the colorScale is [d3.scaleOrdinal](https://observablehq.com/@d3/d3-scaleordinal). Basically, what this does is it associates a specific string (or in our case a value with a specific color.). The values which are not equal with the values specified in the array, will become a darker/lighter shade of the colors.
+        In the following example the colorScale object is defined as `WafermapColorsScale(['red', 'blue', 'green'], ['1', '2', '8']);`\
+        The generated wafer using this color scale is: ![color_scale](./Resources/color_scale.png)
+    -   `dieCharacterCount` - represents the number of characters allowed to be displayed within a single die. As the die values are represented by Floating point numbers, we must have the liberty of limiting how many characters we are willing to display within a single die.
+    -   disabled - it's represented by a boolean value and reffers to the state of the `nimble-wafer-map` component. If true, the component should be rendered dimmed out and no user interaction should be allowed
+
+The `quadrant`, `orientation`, `dieCharacterCount` and `disabled` properties will be configurable via properties and attributes.
+The `die` and `colorScale` properties will be configurable only via properties and will not have attributes.
+
+Methods: The following methods will be exposed in the public API:
+
+-   disable() - renders the component dimmed out and turns off interactivity.
+-   focus() - adds a green box around the component when it has tab focus.
+
+Events: The events mentioned below will all be handled internally by the nimble component and they will not be part of the public API. In the initial implementation the following events should be handled:
+
+-   Zoom in while hovering - this action gets executed whenever the mouse pointer hovers the `nimble-wafer-map` component and a wheel event (scroll up) gets triggered
+-   Zoom out while hovering - this action gets executed whenever the mouse pointer hovers the `nimble-wafer-map` component and a wheel event (scroll down) gets triggered
+-   Drag while zoomed - this event gets triggered whenever the `nimble-wafer-map` component is zoomed in (larger than 100%) and whilst the left mouse button is held the pointer moves to any direction within the wafer-map canvas
+-   Mouse hover - this event gets triggered whenever the mouse pointer is hovering any of the die elements within the wafer map. We only must detect this in the nimble component, proper handling will be done in the MicroStrategy wrapper. (Tooltip triggering)
+
+### Anatomy
+
+Shadow DOM:
+
+```html
+<template id="WaferMapTemplate">
```

Comment:
nit: in practice we would not set an `id` on the `template` element. That would be reflected outside of the shadowroot which would not be expected behavior (the user sets their own `id` on elements). I don't think this will be a major concern, just wanted to point it out.

Reply:
I removed the ID so it appears as the expected layout in the spect.